### PR TITLE
ISPN-5903 Spring split

### PR DESCRIPTION
**Pull request by slaskawi**
Hey!

Please take a look at proposed solution for https://issues.jboss.org/browse/ISPN-5903

This PR is very similar to CDI split implemented a week ago: https://github.com/infinispan/infinispan/pull/3775

Thanks
Sebastian

### CODE REVIEW

File: spring/spring4/spring4-common/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Infinispan Spring support common module
+====================================
+
+This module is not meant to be used individually. It only contains common to Remote and Embedded classes which are attached as sources to those modules.
```
Comment (tristantarrant) [inline spring/spring4/spring4-common/README.md]: "It only contains classes common to ..."

**Comment by slaskawi** [comment]
@tristantarrant done

File: pom.xml [hunk 2]
```diff
@@ -56,7 +56,9 @@
       <module>rhq-plugin</module>
       <module>osgi</module>
       <module>spring/spring</module>
-      <module>spring/spring4</module>
```
Comment (tristantarrant) [inline pom.xml]: Some inconsistency: spring/spring4/pom.xml is never referenced directly by any reactor and therefore it doesn't get "installed" into the repository

Comment (slaskawi) [inline pom.xml]: Yes, that's intentional. `spring/spring4/pom.xml` is a parent pom for Spring 4 modules which doesn't need to be built at all. It just makes things a little bit easier when importing the project into IDE.

Comment (tristantarrant) [inline pom.xml]: Unfortunately, if I then create an application which depends on spring4-embedded it won't find the parent pom in the repository

Comment (slaskawi) [inline pom.xml]: Good point - I haven't thought about it... Fix is right away.

**Comment by slaskawi** [comment]
@tristantarrant done

**Comment by tristantarrant** [comment]
Can you squash the commits please ?

**Comment by slaskawi** [comment]
Yeah sure, any further comments or squashing and are good to go?

**Comment by tristantarrant** [comment]
Looks good

**Comment by tristantarrant** [comment]
Pushed, thanks

**Pull request closed without merging**